ethexe/runtime/common/src/state.rs

```diff
@@ -385,9 +385,17 @@ impl MaybeHashOf<MemoryPages> {
     }
 }
 
-impl MaybeHashOf<MessageQueue> {
+#[derive(Clone, Debug, Decode, Encode, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct MessageQueueHashWithSize {
```

Comment:
I'd make it general primitive: HashOf<T>, SizedHashOf<T>, MaybeHashOf<T>, SizedMaybeHashOf<T>

maybe place todo for it?

Reply:
Added TODO